src/lib/base.jl

```diff
@@ -137,6 +137,10 @@ end
     return getfield(p, i), pair_getfield
 end
 
+@adjoint function Base.Iterators.Zip(is)
+  Base.Iterators.Zip(is), Δ -> (unzip(Δ),)
+end
```

Comment:
```suggestion
@adjoint function Base.Iterators.Zip(is)
  Zip_pullback(Δ) = (unzip(Δ),)
  return Base.Iterators.Zip(is), Zip_pullback
end
```
so that the `@code_warntype` is readable.